src/NexusMods.App/Commandline/CleanupVerbs.cs

```diff
@@ -0,0 +1,180 @@
+                loggingSettings.SlimProcessLogFilePath.ToPath(fileSystem),
+            ];
+
+            var appDirectories = new[]
+            {
+                dataModelSettings.MnemonicDBPath.ToPath(fileSystem),
+                fileExtractorSettings.TempFolderLocation.ToPath(fileSystem),
+                LoggingSettings.GetLogBaseFolder(OSInformation.Shared, fileSystem),
+
+                // Note: This references backend directly in case we ever have
+                // switching backends out. At that point you'd add others here too.
+                JsonStorageBackend.GetConfigsFolderPath(fileSystem),
+
+                // The whole base DataModel folder.
+                DataModelSettings.GetStandardDataModelFolder(fileSystem)
+            }.Concat(dataModelSettings.ArchiveLocations.Select(path => path.ToPath(fileSystem)));
+
+            if (fileSystem.OS.IsUnix())
+                await DeleteRemainingFilesUnix(renderer, appFiles, appDirectories);
+            else
+                DeleteRemainingFilesWindows(appFiles, appDirectories);
+
+            await renderer.Text("Application uninstalled successfully");
+            return 0;
+        }
+        catch (Exception ex)
+        {
+            await renderer.Error(ex, $"Error deleting application directories: {ex.Message}");
+            return -1;
+        }
+        finally
+        {
+            Environment.Exit(0);
+        }
+    }
+
+    private static async Task DeleteRemainingFilesUnix(IRenderer renderer, AbsolutePath[] appFiles, IEnumerable<AbsolutePath> appDirectories)
+    {
+        await DeleteFilesUnix(renderer, appFiles);
+        await DeleteDirectoriesUnix(renderer, appDirectories);
+    }
+    
+    private static void DeleteRemainingFilesWindows(AbsolutePath[] appFiles, IEnumerable<AbsolutePath> appDirectories)
+    {
+        var scriptContent = new StringBuilder();
+
+        // Kill the App Client and Server
+        scriptContent.AppendLine("Stop-Process -Name \"NexusMods.App\" -Force -ErrorAction SilentlyContinue");
+
+        // Note(Sewer) Ensure the process handles are freed, just in case.
+        scriptContent.AppendLine("Start-Sleep -Seconds 1");
+
+        foreach (var appFile in appFiles)
+            scriptContent.AppendLine($"Remove-Item -Path \"{appFile}\" -Force -ErrorAction SilentlyContinue");
+
+        foreach (var directory in appDirectories)
+            scriptContent.AppendLine($"Remove-Item -Path \"{directory}\" -Recurse -Force -ErrorAction SilentlyContinue");
+
+        // Remove the script itself
+        scriptContent.AppendLine("Remove-Item -Path $MyInvocation.MyCommand.Path -Force -ErrorAction SilentlyContinue");
+
+        // Save the script to a temporary file
+        var tempFile = Path.GetTempFileName();
+        var tempScriptPath = tempFile.Replace(".tmp", ".ps1");
+        File.Delete(tempFile);
+        File.WriteAllText(tempScriptPath, scriptContent.ToString());
```

Comment:
This should just be a file we ship, no need to generate this at runtime.

Reply:
There's actually a reason I wrote it this way.

The parameters `appFiles` and `appDirectories` are non-deterministic. Many of them can be customized by the user, or will be customizable by the user. For example, locations of `.nx` archives.

Therefore, if I were to ship this as a static file, I'd still need to pass locations to the script via parameters.

The edge case you get here is that the NT API uses [UNICODE_STRING](https://learn.microsoft.com/en-us/windows/win32/api/ntdef/ns-ntdef-_unicode_string) to start processes. The max length of that is technically 32K characters, however that is shared with some things such as the `Environment` block. In Microsoft's own software, such as `cmd`, they often therefore restrict the command length to `8191`.

Assuming `8191` as a 'safe' limit, passing all the paths as parameters yields a non-zero chance that the sum of all the paths is long enough that the length limit is exceeded. Because:

- The user could set a very long path to one of their locations. 
- We add more configurable locations to the App.
- The user sets a lot of locations where the limit is not bounded.
    - e.g. `Nx` archives can be set to an arbitrary number of locations.

If it happens that the command length is too long, the failure would (most likely) be silent, and thus no cleanup would be performed. That's problematic, as we'd then possibly never get a bug report for it.

That's why I chose to inline the deletes into the script.

Yes. The probability of this of course is all very low; a 'super edge case', but it still exists.
I can add this as a note to the code, however this is the reason why I chose to do it this way.

Reply:
Make a static script and write the paths to a file in that case. Let the script iterate over each line in the generate text file to delete the directories.

The reason I'm so pedantic about this, is because generating powershell scripts at runtime is a big no no in terms of security. We have no idea what weird antivirus programs will do when they see this.

Reply:
Pretty valid concern. Historically it hasn't caused me issues, but I might aswell.

I'll get that sorted after PR review.